majilove/models.py

```diff
@@ -106,3 +108,402 @@ def display_blood_type(self):
+    icon_special_shot = models.ImageField(string_concat(_('Icon'), ' (', _('Special Shot'), ')'), upload_to=uploadItem('p/a'), null=True)
+
+    # Full photo
+    image = models.ImageField(_('Photo Image'), upload_to=uploadItem('p/image'))
+    image_special_shot = models.ImageField(string_concat(_('Photo Image'), ' (', _('Special Short'), ')'), upload_to=uploadItem('p/image/a'), null=True)
+
+    transparent = models.ImageField(_('Transparent'), upload_to=uploadItem('p/transparent'), null=True)
+    transparent_special_shot = models.ImageField(string_concat(_('Transparent'), ' (', _('Special shot'), ')'), upload_to=uploadItem('p/transparent/a'), null=True)
+
+    poster = models.ImageField(_('Poster'), upload_to=uploadItem('p/poster'), null=True)
+    poster_special_shot = models.ImageField(string_concat(_('Poster'), ' (', _('Special Shot'), ')'), upload_to=uploadItem('p/poster/a'), null=True)
+
+    message = models.ImageField(_('Message'), upload_to=uploadItem('p/message'), null=True)
+    autograph = models.ImageField(_('Autograph'), upload_to=uploadItem('p/autograph'), null=True)
+
+    message_text = models.TextField(string_concat(_('Message Text'), ' (', _('Japanese') + ')'), max_length=500, null=True)
+    message_translation = models.TextField(_('Message translation'), max_length=500, null=True)
+    MESSAGE_TRANSLATIONs_CHOICES = ALL_ALT_LANGUAGES
+    d_message_translations = models.TextField(_('Message translation'), null=True)
+    @property
+    def t_message_translation(self):
+        if get_language() == 'ja': return None
+        return self.message_translations.get(get_language(), self.message_translation)
+
+    # photo stats
+
+    #triplets_in_moments is number of sets of +30 dance/vocal/charm sqaures before 100%
+    RARITIES = OrderedDict([
+        (1, {
+            'short_form': 'N',
+            'special_shot_percentage': None,
+            'outfit_unlock_percentage': 0,
+            'triplets_in_moments' : 1,
+            'max_levels': 20,
+            }),
+        (2, {
+            'short_form': 'R',
+            'special_shot_percentage': 100,
+            'outfit_unlock_percentage': 50,
+            'triplets_in_moments' : 4,
+            'max_levels': (30, 50),
+            }),
+        (3, {
+            'short_form': 'SR',
+            'special_shot_percentage': 83,
+            'outfit_unlock_percentage': 33,
+            'triplets_in_moments' : 6,
+            'max_levels': (40, 60),
+            }),
+        (4, {
+            'short_form': 'UR',
+            'special_shot_percentage': 87,
+            'outfit_unlock_percentage': 25,
+            'triplets_in_moments': 8,
+            'max_levels': (50, 70),
+        }),
+    ])
+
+    RARITY_CHOICES = [(_name, _info['short_form']) for _name, _info in RARITIES.items()]
+    RARITY_WITHOUT_I_CHOICES = True
+    i_rarity = models.PositiveIntegerField(_('Rarity'), choices=i_choices(RARITY_CHOICES), db_index=True)
```

Comment:
I'd recommend using `getInfoFromChoices` to retrieve the various details depending on the rarity, which you could then use below.

```python
i_rarity = ...
rarity_max_levels = property(getInfoFromChoices('rarity', RARITIES, 'max_levels'))
rarity_special_shot_percentage = ... # same
```